src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -116,7 +116,7 @@ private ushort Significand
                 // says they should be equal, even if the signs differ.
                 return leftIsNegative && !AreZero(left, right);
             }
-            return (short)(left._value) < (short)(right._value);
+            return left._value < right._value ^ leftIsNegative;
```

Comment:
Just for reference, here is the lt implementation: https://github.com/ucb-bar/berkeley-softfloat-3/blob/master/source/f16_lt_quiet.c

Reply:
Looks like it's pretty much the same.